src/pretix/plugins/webcheckin/static/pretixplugins/webcheckin/components/app.vue

```diff
@@ -360,6 +360,9 @@ export default {
                 reason: 'invalid',
               }
             }
+            if (!response.ok && [401, 403].includes(response.status)) {
+              window.location.href = '/control/login?next=' + window.location.pathname + window.location.hash;
```

Comment:
you probably need to call escape() on pathname and hash for the hash to actually work. are we sure we never have a query string here?

also, can we imagine reasons for 401/403 where the redirect does not make sense?

Reply:
I tried it out like this and it worked as intended...

Not sure about if there could be 401 or 403 that do not warrant a re-login... Possibly if the user got its access-rights removed to that specific event... In that case, the relogin wouldn't change anything.

Reply:
AFAIK not encoding e.g. "/" and most other chars works most of the time in current browsers, but it should be encoded nonetheless. See https://datatracker.ietf.org/doc/html/rfc3986#section-2.2

Also I don’t think the hash works as intended – it will not be included in the next-GET-param but will be interpreted as a hash (Sprungmarke?) on the /control/login page. So I think – as Raphael pointed out as well – it should be encoded. So basically:

```suggestion
              window.location.href = '/control/login?next=' + encodeURIComponent(window.location.pathname + window.location.search + window.location.hash);
```

Reply:
> Also I don’t think the hash works as intended

It should not work, but if @pc-coholic  says he tested, I believe that – browser have a weird behaviour of carrying the hash across redirects in some cases.

We should still use `encodeURIComponent` and `window.location.search` to be future-proof, this would be a really annoying bug to find if we missed that

Reply:
> It should not work, but if @pc-coholic says he tested, I believe that – browser have a weird behaviour of carrying the hash across redirects in some cases.

Yes, I believe so too – sorry @pc-coholic if that did not come across.

Reply:
Everything is fine - no worries :-)